shopinvader_api_signin_jwt/routers/signin.py

```diff
@@ -34,22 +34,15 @@ def signin(
     Authenticate the partner based on a JWT token or a session cookie.
     Set the session cookie if allowed.
     Return HTTP code 201 if res.partner created (case of the first signin).
-    Transfer anonymous cart and delete anonymous partner if any.
+    Promote anonymous partner and delete it if any.
     """
     if not partner:
         partner = env[
             "shopinvader_api_signin_jwt.signin_router.helper"
         ]._create_partner_from_payload(payload)
         response.status_code = status.HTTP_201_CREATED
-    anonymous_partner = env["res.partner"]._get_anonymous_partner__cookie(
-        request.cookies
-    )
-    if anonymous_partner:
-        anonymous_cart = env["sale.order"].sudo()._find_open_cart(anonymous_partner.id)
-        if anonymous_cart:
-            anonymous_cart._transfer_cart(partner.id)
-            anonymous_cart.unlink()
-        env["res.partner"]._delete_anonymous_partner__cookie(request.cookies, response)
+
+    env["res.partner"]._promote_anonymous_partner(partner, request.cookies, response)
```

Comment:
```suggestion
    if anonymous_partner:
        partner._promote_from_anonymous_partner(anonymous_partner)
        env["res.partner"]._delete_anonymous_partner__cookie(request.cookies, response)
```

I find this much easier to follow and don't see the benefit of factoring it out.

Reply:
### After discussing with @paradoxxxzero 

I understand why he do not want to duplicate the following code

```
        anonymous_partner = self._get_anonymous_partner__cookie(cookies)
        if anonymous_partner:
            partner._promote_from_anonymous_partner(anonymous_partner)
            env["res.partner"]._delete_anonymous_partner__cookie(cookies, response)
```


Indeed right now the promote is be done
- on signin in jwt auth 
- on signin in partner_auth
- on reset the password in partner auth

In long term we may have additionnal way to signin
- like magic link
- other auth implementation (oauth, social network ...).

So everythere we do the promote we have to put the previous 4 lines


So it make sense to call a method that always do the same thing.

### At the same time

I understand @sbidoul.
The first time, I look at the code I was a little lost with the method "_promote_anonymous_partner" and "_promote_from_anonymous_partner"

It hard to understand what they do without looking deeper in the code

Also it always fell weird for me to have cookie logic in the "res.partner" model


### Maybe we should

move the logic of cookies stuff in a dedicated helper

```
class ShopinvaderAnonymousCookieHelper(models.AbstractModel):
    _name = "shopinvader_anonymous_partner.cookie.helper"

    def _create_anonymous_partner__cookie(self, response: Response):
       ...

    def _delete_anonymous_partner__cookie(self, cookies: Cookies, response: Response):
        ...

    def _create_anonymous_partner__cookie(self, response: Response):
       ...

    def _promote_anonymous_partner_and_delete_cookie(self, partner: ResPartner, cookies: Cookies, response: Response):
        anonymous_partner = self._get_anonymous_partner__cookie(cookies)
        if anonymous_partner:
            partner._promote_from_anonymous_partner(anonymous_partner)
            self._delete_anonymous_partner__cookie(cookies, response)

```

Then on the partner we only have code related to the model, that act on model

```

class ResPartner

    def _create_anonymous_partner__token(self):
        ....
   
    def _get_anonymous_partner__token(self, token: str):
       ....

   def _promote_from_anonymous_partner(self, anonymous_partner: ResPartner):
       ...

```

Then in jwt signin we just do and in all other method that do an auth

```
    helper = env["shopinvader_anonymous_partner.cookie.helper"]
    helper._promote_anonymous_partner_and_delete_cookie(partner, request.cookies, response)
```


What do you think ?

Reply:
It's hard to name thing :cry:

Reply:
I had lost track of this. Sounds like a good idea.